xtask/src/main.rs

```diff
@@ -0,0 +1,69 @@
+use std::{env, process::Command};
+
+type DynError = Box<dyn std::error::Error>;
+
+fn main() {
+    if let Err(e) = execute_task() {
+        eprintln!("{e}");
+        std::process::exit(-1);
+    }
+}
+
+fn execute_task() -> Result<(), DynError> {
+    let task = env::args().nth(1);
```

Comment:
We already depend on Clap, so you could use it here (feel free to do this in a later PR!).

Reply:
Have pushed a new PR (#2707)  for this and the other suggestions.